.github/workflows/dns-preview.yml

```diff
@@ -0,0 +1,29 @@
+---
+name: Test/Preview DNS changes
+
+on:
+  pull_request:
+    paths:
+      - dns/
+
+permissions: {}
+
+jobs:
+  dnscontrol:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4.2.2
+      - uses: cachix/install-nix-action@v31
+      - name: dnscontrol check
+        working-directory: ./dns/
+        run: |
+          nix run --inputs-from . nixpkgs#dnscontrol -- check
```

Comment:
Does this need network? We could make this a static check as well.

Reply:
It does not.